src/main/java/nextstep/courses/domain/session/Session.java

```diff
@@ -2,50 +2,63 @@
 
 import java.util.List;
 import nextstep.courses.domain.BaseEntity;
-import nextstep.courses.domain.image.SessionCoverImage;
+import nextstep.courses.domain.image.SessionCoverImages;
 import nextstep.courses.domain.registration.Registration;
 import nextstep.courses.domain.registration.Registrations;
+import nextstep.courses.domain.session.recruit.Recruiting;
 
 public class Session extends BaseEntity {
     private final Long courseId;
     private final Term term;
     private final SessionPeriod period;
-    private final SessionCoverImage coverImage;
+    private final SessionCoverImages coverImages;
     private final SessionPolicy sessionPolicy;
-    private SessionState state;
+    private SessionProgressState progressState;
 
-    public Session(Long courseId, int term, String startDay, String endDay, SessionCoverImage coverImage) {
-        this(null, courseId, new Term(term), new SessionPeriod(startDay, endDay), SessionState.PREPARING, new SessionPolicy(), coverImage);
+    public Session(Long courseId, int term, String startDay, String endDay, SessionCoverImages coverImages) {
+        this(null, courseId, new Term(term), new SessionPeriod(startDay, endDay), SessionProgressState.PREPARING, new SessionPolicy(), coverImages);
     }
 
-    public Session(Long courseId, int term, String startDay, String endDay, SessionPolicy sessionPolicy, SessionCoverImage coverImage) {
-        this(null, courseId, new Term(term), new SessionPeriod(startDay, endDay), SessionState.PREPARING, sessionPolicy, coverImage);
+    public Session(Long courseId, int term, String startDay, String endDay, SessionPolicy sessionPolicy, SessionCoverImages coverImages) {
+        this(null, courseId, new Term(term), new SessionPeriod(startDay, endDay), SessionProgressState.PREPARING, sessionPolicy, coverImages);
     }
 
-    public Session(Long id, Long courseId, Term term, SessionPeriod period, SessionState state, SessionPolicy sessionPolicy, SessionCoverImage coverImage) {
+    public Session(Long id, Long courseId, Term term, SessionPeriod period, SessionProgressState progressState, SessionPolicy sessionPolicy, SessionCoverImages coverImages) {
         super(id);
         this.courseId = courseId;
         this.term = term;
         this.period = period;
-        this.state = state;
+        this.progressState = progressState;
         this.sessionPolicy = sessionPolicy;
-        this.coverImage = coverImage;
+        this.coverImages = coverImages;
+    }
+    public Enrollment enrollment(Registration registration) {
+      validateCanCreateEnrollment();
+      return new Enrollment(this.getId(), new Recruiting(), sessionPolicy, new Registrations(registration));
```

Comment:
```suggestion
      return enrollment(new Registrations(registration));
```
메서드를 오버로딩할 경우 오버로딩하는 다른 메서드를 호출하면 중복도 제거할 수 있음.
생성자의 주/부와 같은 효과